src/binarySearchTree.js

```diff
@@ -0,0 +1,95 @@
+import BinaryNode from './binarynode'
+
+export default class BinarySearchTree {
+  constructor() {
+    this.length = 0
+    this.root = null
+  }
+
+  insert = (value) => {
+    const node = new BinaryNode(value)
+    if (this.count() == 0) {
+      this.root = node
+      this.length += 1
+    }
+    else {
+      let currentNode = this.root
+      for(let i = 0; i < this.count(); i++){
+        if(currentNode._value < node._value){
+          if(currentNode._rightChild == null){
+            currentNode._rightChild = node
+            this.length += 1
+            return
+          }
+          currentNode = this[i]._rightChild
+        }
+        if(currentNode._value > node._value){
+          if(currentNode._leftChild == null){
+            currentNode._leftChild = node
+            this.length += 1
+            return
+          }
+          currentNode = this[i]._leftChild
+        }
+      }
+    }
+  }
+
+  search = (value) => {
+    if(this.count() === 0){
+      return null
+    }
+    let currentNode = this.root
+    for(let i = 0; i < this.count(); i++){
+      if(currentNode._value == value){
+        return currentNode
+      }
+      if(currentNode._value < value){
+        currentNode = currentNode._rightChild
+      }
+      if(currentNode._value > value){
+        currentNode = currentNode._leftChild
+      }
+    }
+    return null
+  }
+
+  remove = (value) => {
+    if(this.count() === 0){
+      return null
+    }
+    const node = new BinaryNode(value)
+    if(this.root._value === value)
+    {
+      if(this.count() === 1){
+        this.root = null
+        this.length -= 1
+        return
+      }
+      else{
```

Comment:
For multi-line code-blocks, put else on the same line as your if's closing brace `} else {`. 
A great style-guide to use as reference is [air-bnb's](https://github.com/airbnb/javascript)